lib/ansible/modules/net_tools/ipwcli_dns.py

```diff
@@ -0,0 +1,363 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: ipwcli_dns
+
+short_description: Manage DNS Records for Ericsson IPWorks via ipwcli
+
+version_added: "2.10"
+
+description:
+    - "Manage DNS records for the Ericsson IPWorks DNS server. The module will use the ipwcli to deploy the DNS records."
+
+requirements:
+   - ipwcli (installed on Ericsson IPWorks)
+
+notes:
+    - To make the DNS record changes effective, you need to run 'update dnsserver' on the ipwcli.
+
+options:
+    dnsname:
+        description:
+            - Name of the record.
+        required: true
+        type: str
+    type:
+        description:
+            - Type of the record.
+        required: true
+        type: str
+        choices: [ NAPTR, SRV, A, AAAA ]
+    container:
+        description:
+            - Sets the container zone for the record.
+        required: true
+        type: str
+    address:
+        description:
+            - The IP address for the A or AAAA record.
+            - Required for C(type=A) or C(type=AAAA)
+        type: str
+    ttl:
+        description:
+            - Sets the TTL of the record.
+        type: int
+        default: 3600
+    state:
+        description:
+            - Whether the record should exist or not.
+        type: str
+        choices: [ absent, present ]
+        default: present
+    priority:
+        description:
+            - Sets the priority of the SRV record.
+        type: int
+        default: 10
+    weight:
+        description:
+            - Sets the weight of the SRV record.
+        type: int
+        default: 10
+    port:
+        description:
+            - Sets the port of the SRV record.
+            - Required for C(type=SRV)
+        type: int
+    target:
+        description:
+            - Sets the target of the SRV record.
+            - Required for C(type=SRV)
+        type: str
+    order:
+        description:
+            - Sets the order of the NAPTR record.
+            - Required for C(type=NAPTR)
+        type: int
+    preference:
+        description:
+            - Sets the preference of the NAPTR record.
+            - Required for C(type=NAPTR)
+        type: int
+    flags:
+        description:
+            - Sets one of the possible flags of NAPTR record.
+            - Required for C(type=NAPTR)
+        type: str
+        choices: ['S', 'A', 'U', 'P']
+    service:
+        description:
+            - Sets the service of the NAPTR record.
+            - Required for C(type=NAPTR)
+        type: str
+    replacement:
+        description:
+            - Sets the replacement of the NAPTR record.
+            - Required for C(type=NAPTR)
+        type: str
+    user:
```

Comment:
The more common variant for this option name is
```suggestion
    username:
```

Reply:
updated